### PR TITLE
feat(TX-681): remove modal from checkout + fix modal header spacing

### PR DESCRIPTION
**Pull request by rquartararo**
### Description
This PR:
1. Removes the modal view from the BNMO checkout flow
2. Removes the extra spacing above the modal headers for Admin and all modals using `ArtsyViewView`. Please let me know if you know of any other modals that are experiencing the extra header spacing issue.  

This PR resolves [TX-681] and [TX-633]

#### Modal removal from checkout flow
https://user-images.githubusercontent.com/50849237/190136836-f7866c89-e829-4d51-93af-6b0b21f6146e.mp4

#### Extra modal header spacing before and after
Simulator Screen Shot - iPhone 11 - 2022-09-14 at 12 13 20

Simulator Screen Shot - iPhone 11 - 2022-09-14 at 12 13 10

#### Acceptance Criteria

-

#### Setup Instructions

-

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- removed modal view from BNMO checkout flow

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[TX-681]: https://artsyproduct.atlassian.net/browse/TX-681?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TX-633]: https://artsyproduct.atlassian.net/browse/TX-633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AlicanAkyuz — APPROVED** [review]
lgtm!

File: src/app/Components/ArtsyWebView.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const ArtsyWebViewPage: React.FC<
   backProps,
   backAction,
 }) => {
-  const paddingTop = useScreenDimensions().safeAreaInsets.top
+  const paddingTop = isPresentedModally ? "0" : useScreenDimensions().safeAreaInsets.top
```
Comment (pvinis) [inline src/app/Components/ArtsyWebView.tsx:68]:
```suggestion
  const paddingTop = isPresentedModally ? 0 : useScreenDimensions().safeAreaInsets.top
```

tiiiiiiiny nothing

File: src/app/Scenes/Artwork/Components/CommercialButtons/MakeOfferButton.tsx [hunk 2]
```diff
@@ -115,7 +115,6 @@ export class MakeOfferButton extends React.Component<MakeOfferButtonProps, State
                 this.onMutationError(orderOrError.error)
               } else if (orderOrError.__typename === "CommerceOrderWithMutationSuccess") {
                 navigate(`/orders/${orderOrError.order.internalID}`, {
-                  modal: true,
```
Comment (pvinis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/MakeOfferButton.tsx:118]: i wonder if we use this anywhere else that we could remove 🤔

**Review by pvinis — APPROVED** [review]
nice!

File: src/app/Scenes/Artwork/Components/CommercialButtons/MakeOfferButton.tsx (hunk 2 above)
Comment (pvinis) [reply src/app/Scenes/Artwork/Components/CommercialButtons/MakeOfferButton.tsx:118]: indeed, also i found we use it for markdown-rendered stuff in descriptions. cool, so we keep that

File: src/app/Components/ArtsyWebView.tsx (hunk 1 above)
Comment (pvinis) [reply src/app/Components/ArtsyWebView.tsx]: you can totally just click the commit button on github, instead of doing the change locally :)

**Review by pvinis — APPROVED** [review]

**Pull request merged**